src/App.tsx

```diff
@@ -1,61 +1,41 @@
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { useState } from 'react';
+import { Todo } from './type/Todo';
+import { NewTodo, TodoCreateData } from './components/NewTodo';
+
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const preparedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <form action="/api/todos" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>
-              Choose a user
-            </option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
+  const [todos, setTodos] = useState<Todo[]>(preparedTodos);
 
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
+  const handleAddTodo = (data: TodoCreateData) => {
+    const todo = {
+      user: getUserById(data.userId),
+      id: Math.max(...todos.map(item => item.id)) + 1,
+      title: data.title,
+      completed: false,
+      userId: data.userId,
+    };
 
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
+    setTodos([...todos, todo]);
```

Comment:
When you need to set state depending on the previous one use function `setState`